transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,229 @@
-import csv
-import io
-from typing import Set, Tuple
+from datetime import timedelta
+from typing import Dict, Optional
 
+from transit_odp.common.csv import CSVBuilder, CSVColumn
 from transit_odp.organisation.models import TXCFileAttributes
+from transit_odp.organisation.models.data import SeasonalService
 from transit_odp.otc.models import Service as OTCService
+from transit_odp.publish.requires_attention import (
+    get_otc_map,
+    get_txc_map,
+    requires_attention,
+    requires_attention_data,
+)
 
+STALENESS_STATUS = [
+    "Stale - End date passed",
+    "Stale - 12 months old",
+    "Stale - OTC Variation",
+]
 
-class ServiceCodesCSV:
-    REGISTERED = "Registered"
-    MISSING_DATA = "Missing data"
+
+def add_seasonal_service_status(otc_service: dict) -> str:
+    seasonal_service_status = otc_service.get("seasonal_status")
+    seasonal_service_start = otc_service.get("seasonal_start")
+    return (
+        "In Season"
+        if seasonal_service_status
+        else "Not Seasonal"
+        if seasonal_service_start is None
+        else "Out of Season"
+    )
+
+
+def get_seasonal_service_map(organisation_id: int) -> Dict[str, SeasonalService]:
+    """
+    Get a list of dictionaries which includes all the Seasonal Services
+    for an organisation.
+    """
+    return {
+        service.registration_number.replace("/", ":"): service
+        for service in SeasonalService.objects.filter(
+            licence__organisation_id=organisation_id
+        )
+        .add_registration_number()
+        .add_seasonal_status()
+    }
+
+
+class ServiceCodesCSV(CSVBuilder):
+    columns = [
+        CSVColumn(
+            header="Requires Attention",
+            accessor=lambda otc_service: otc_service.get("require_attention"),
+        ),
+        CSVColumn(
+            header="Published Status",
+            accessor=lambda otc_service: "Published"
+            if otc_service.get("dataset_id")
+            else "Unpublished",
+        ),
+        CSVColumn(
+            header="OTC status",
+            accessor=lambda otc_service: "Registered"
+            if otc_service.get("licecnce_number")
+            else "Unregistered",
+        ),
+        CSVColumn(
+            header="Scope Status",
+            accessor=lambda otc_service: "Out of Scope"
+            if otc_service.get("service_code") and not otc_service.get("dataset_id")
+            else "In Scope",
+        ),
```

Comment:
Scope Status is about whether the service code is exempted in the service code exemption workflow. I don't see how this logic captures that.